src/decomp_2d_init_fin.f90

```diff
@@ -47,34 +47,12 @@ subroutine decomp_2d_init_ref(nx, ny, nz, p_row, p_col, periodic_bc, comm)
      if (decomp_profiler_d2d) call decomp_profiler_start("decomp_2d_init")
 #endif
 
-     ! Use the provided MPI communicator if present
-     if (present(comm)) then
-        decomp_2d_comm = comm
-     else
-        decomp_2d_comm = MPI_COMM_WORLD
-     end if
+     call decomp_2d_mpi_init(comm)
 
-     ! Safety check
```

Comment:
This was removed, it should be harmless to keep it

Reply:
Yes I’ll put it back, but in `decomp_2d_mpi_init` under the `constants` module.